llumnix/llm_engine_manager.py

```diff
@@ -211,37 +210,39 @@ async def _clear_request_instance_loop(self, interval: float):
             await asyncio.sleep(interval)
             self.request_instance = {}
 
-    async def _post_migrate(self, rets: List[str], call_migrate_instance_pairs: List[Tuple[str, str]]) -> None:
-        for i, ret in enumerate(rets):
-            self.instance_migrating[call_migrate_instance_pairs[i][0]] = False
-            self.instance_migrating[call_migrate_instance_pairs[i][1]] = False
+    async def _migrate(self) -> None:
+        async def migrate_done_callback(ret, migrate_instance_pair: Tuple[str, str]) -> None:
+            self.instance_migrating[migrate_instance_pair[0]] = False
+            self.instance_migrating[migrate_instance_pair[1]] = False
             if isinstance(ret, (ray.exceptions.RayActorError, KeyError)):
-                has_error_pair = await self._check_instance_error(call_migrate_instance_pairs[i])
-                for j, has_error in enumerate(has_error_pair):
+                has_error_pair = await self._check_instance_error(migrate_instance_pair)
+                for i, has_error in enumerate(has_error_pair):
                     # Instance without error should clear migration states.
                     if not has_error:
                         try:
-                            await self.instances[call_migrate_instance_pairs[i][j]].clear_migration_states.remote(is_migrate_in=bool(j))
+                            await self.instances[migrate_instance_pair[i]].clear_migration_states.remote(is_migrate_in=bool(i))
                         except (ray.exceptions.RayActorError, KeyError):
                             has_error = True
-                for j, has_error in enumerate(has_error_pair):
+                for i, has_error in enumerate(has_error_pair):
                     if has_error:
-                        instance_id = call_migrate_instance_pairs[i][j]
+                        instance_id = migrate_instance_pair[i]
                         logger.info("[_migrate] instance {} is dead".format(instance_id))
                         self.scale_down(instance_id)
             else:
                 migrate_out_request_ids = ret
                 if migrate_out_request_ids:
                     migrate_out_request_id = migrate_out_request_ids[0]
-                    self.request_instance[migrate_out_request_id] = call_migrate_instance_pairs[i][1]
+                    self.request_instance[migrate_out_request_id] = migrate_instance_pair[1]
                 logger.info("{}->{} migrate done, migrate request {}".format(
-                    call_migrate_instance_pairs[i][0], call_migrate_instance_pairs[i][1], migrate_out_request_ids))
+                    migrate_instance_pair[0], migrate_instance_pair[1], migrate_out_request_ids))
+        def migrate_done_callback_wrapper(migrate_instance_pair: Tuple[str, str], fut) -> None:
+            ret = fut.result()
```

Comment:
exception maybe need handle here ?

Reply:
exception is returned, and is tested in callback.